### PR TITLE
Cast number64 to json number in export plugin

### LINKED ISSUE #321: Json values with ULL suffix
Some metrics parse their values to ULL strings:
```json
[{
   "label_pairs":{
      "engine":"memtx",
      "name":"MY_SPACE",
      "alias":"tnt_router"
   },
   "timestamp":1635867200499883,
   "metric_name":"tnt_space_bsize",
   "value":"51589ULL"
},
{
   "label_pairs":{
      "engine":"memtx",
      "name":"MY_SPACE",
      "alias":"tnt_router"
   },
   "timestamp":1635867200499883,
   "metric_name":"tnt_space_total_bsize",
   "value":"199045ULL"
}]
```
This values can't be parsed by Telegraf+InfluxDB monitoring stack. This issue was already solved for Prometheus (https://github.com/tarantool/metrics/issues/15) and Graphite (https://github.com/tarantool/metrics/issues/90), but not for json.

### PR DESCRIPTION
**Pull request by DifferentialOrange**
Closes #321

### CODE REVIEW

File: metrics/plugins/json.lua [hunk 1]
```diff
@@ -6,6 +6,11 @@ local function finite(value)
     if type(value) == "string" then
         value = tonumber(value)
         if value == nil then return nil end
+    elseif type(value) == "cdata" then
+        value = tostring(value)
+        -- Luajit cdata type inserts some postfix in the end of the number after tostring() operation
+        value = value:gsub("U*LL", "")
```
Comment (Totktonada) [inline metrics/plugins/json.lua:12]: Nit: we can use `U?LL` here.

File: metrics/plugins/json.lua [hunk 2]
```diff
@@ -6,6 +6,11 @@ local function finite(value)
...
+        return tonumber(value)
```
Comment (Totktonada) [inline metrics/plugins/json.lua:13]: How about numbers out of the range [-2^53; 2^53] range, where double (lua number) types cannot represent all integral numbers? There is `number64()` function. BTW, it returns just `number` if the value is small enough.

(I'm not in context here, but, it seems, this part of the code may be just `number64()` call.)

File: test/plugins/json_test.lua [hunk 3]
```diff
@@ -58,3 +58,21 @@ g.test_number_value_serialization = function()
     t.assert_equals(float_obs.value, 0.333, 'check float')
     t.assert_equals(int_obs.value, 10, 'check int')
 end
+
+g.test_cdata_ll_value_parses_to_number = function()
+    local gauge_ll = metrics.gauge('test_cdata_ll')
+    gauge_ll:set(-205LL)
```
Comment (Totktonada) [inline test/plugins/json_test.lua:64]: number64 values (cdata<int64_t> / cdata<uint64_t>) usually occurs when you working with very large (or very small negative) integral numbers. So I think that we should test those cases.

Comment (DifferentialOrange) [reply test/plugins/json_test.lua:64]: Well, I've got ULL suffix in `51589ULL`. But I can increase the number

File: metrics/plugins/json.lua (hunk 2 above)
Comment (DifferentialOrange) [reply metrics/plugins/json.lua:13]: Oh, thank you. I've based my solution on https://github.com/tarantool/metrics/pull/23 , but I think it is worth to use more simpler one

File: test/plugins/json_test.lua (hunk 3 above)
Comment (DifferentialOrange) [reply test/plugins/json_test.lua]: Changed

File: metrics/plugins/json.lua (hunk 2 above)
Comment (DifferentialOrange) [reply metrics/plugins/json.lua:10]: Reworked

File: metrics/plugins/json.lua (hunk 1 above)
Comment (DifferentialOrange) [reply metrics/plugins/json.lua]: There is no more gsub

File: metrics/plugins/json.lua (hunk 2 above)
Comment (Totktonada) [reply metrics/plugins/json.lua:10]: I still don't understand why `tonumber()` is there. It'll break some numbers outside of the [-2^53; 2^53] range:

```lua
tarantool> tonumber(9007199254740993ULL)
---
- 9007199254740992
...

tarantool> tonumber64(9007199254740993ULL)
---
- 9007199254740993
...
```

File: test/plugins/json_test.lua (hunk 3 above)
Comment (Totktonada) [reply test/plugins/json_test.lua]: `51589ULL` is less than 10^14 (where we started to encode it into cdata) and less than 2^53, where we can loss precision below 1.0. We should check that, say `9007199254740992ULL` (2^53) and numbers around it are marshalled correctly.

You can consider this following illustration how integral numbers are marshalled in tarantool:

https://github.com/tarantool/conf/blob/80078f85423ffd54d812c11b47aa85729e9e3d87/conf/client/etcd/scalar_serializer.lua#L134-L152

File: metrics/plugins/json.lua (hunk 2 above)
Comment (DifferentialOrange) [reply metrics/plugins/json.lua:10]: Well, dodging precision loss is impossible until we support encoding cdata to numbers in json package. Metric value as string is not supported by Telegraf/InfluxDB, who is the main (if not the only one) consumer of json metrics.

File: test/plugins/json_test.lua (hunk 3 above)
Comment (DifferentialOrange) [reply test/plugins/json_test.lua]: Looks like space bsize is always cdata
```
Tarantool 2.10.0-beta1-0-g7da4b1438
type 'help' for interactive help
tarantool> box.cfg{}
---
...
tarantool> box.space._space:bsize()
---
- 4151
...

tarantool> type(box.space._space:bsize())
---
- cdata
...
```

Comment (Totktonada) [reply test/plugins/json_test.lua]: (Tarantool 2.10.0-beta1-91-g16a484a30.)

```
tarantool> print(json.encode(box.space._space:bsize()))
7943
---
...
```

Do you see another behaviour of the json encoder on another tarantool version?

File: metrics/plugins/json.lua (hunk 2 above)
Comment (DifferentialOrange) [reply metrics/plugins/json.lua]: Seems that I am wrong about cdata transform.

File: test/plugins/json_test.lua (hunk 3 above)
Comment (DifferentialOrange) [reply test/plugins/json_test.lua]: Actually no, it is as you described. Looks like I was wrong. Thank you! I had reworked json export to be as simple as possible.

File: metrics/plugins/json.lua (hunk 2 above)
Comment (Totktonada) [reply metrics/plugins/json.lua]: Okay.

File: test/plugins/json_test.lua (hunk 3 above)
Comment (Totktonada) [reply test/plugins/json_test.lua]: Okay so.

**Comment by Totktonada** [comment]
I don't know, whether you expect a review from me. Actually I never work with metrics code, so I would prefer to don't review. But all my comments are resolved now.

**Comment by DifferentialOrange** [comment]
@yngvar-antonsson , can you review this one?

**Review by yngvar-antonsson — APPROVED** [review]

**Pull request merged**